src/sentry/auth/helper.py

```diff
@@ -969,3 +970,24 @@ def disable_2fa_required(self) -> None:
             event=audit_log.get_event_id("ORG_EDIT"),
             data={"require_2fa": "to False when enabling SSO"},
         )
+
+
+@transaction.atomic
+def EnablePartnerSSO(provider_key, org_member, sentry_org, provider_config):
+    provider_model = AuthProvider.objects.create(
+        organization_id=sentry_org.id, provider=provider_key, config=provider_config
+    )
+
+    sso_enabled.send_robust(
+        organization=sentry_org,
+        provider=provider_key,
+        sender="EnablePartnerSSO",
+    )
+
+    AuditLogEntry.objects.create(
+        organization_id=sentry_org.id,
+        actor=org_member,
```

Comment:
Who is this org_member at the time of provisioning?

Reply:
This requires us to pass in an appropriate OrgMember instance.

My understanding is that at provisioning time we'll have created both a new organization as well as an OrgMember instance with no associated user identity.

Reply:
Discussed offline. Let's test this in prod :P